Tensile/Utilities/toFile.py

```diff
@@ -37,10 +40,25 @@ def toFile(outputFile: Path, contents: List[str], delimiter: str = "\n") -> None
     
     Raises:
         AssertionError: If contents is not a List[str]
+
+    Notes:
+        - If outputFile is a TextIOWrapper, the file must not be closed.
+        - Providing an open file is useful for writing multiple calls to the same file.
     """
-    assert isinstance(contents, list), "contents must be a list."
-    assert isinstance(contents[0], str), "contents elements must be a str."
+    assert isinstance(contents, list), f"contents must be a list, found {type(contents)}"
+    assert isinstance(contents[0], str), f"contents elements must be a str, found {type(contents[0])}"
+
+    isTextIO = isinstance(outputFile, TextIOWrapper)
+    if isTextIO:
+        if outputFile.closed:
+            raise ValueError(f"Provided file {outputFile.name} is already closed: outputFile must be an open file, or a Path.")
+
+    f = outputFile if isTextIO else open(outputFile, "w")
 
-    with open(outputFile, "w") as generatedFile:  
-      for filePath in contents:
-        generatedFile.write(f"{filePath}{delimiter}")
+    for chunk in contents:
+        f.write(f"{chunk}{delimiter}")
```

Comment:
If this fails and we aren't a TextIO then we leak a reseource.

Reply:
The re-usability of this function is quite nice IMO, so another option would be to use a try/except and handle any failures here gracefully.